restic/cfg/config.go

```diff
@@ -0,0 +1,91 @@
+package cfg
+
+import (
+	"fmt"
+	"strings"
+)
+
+const (
+	// RestoreTypeS3 indicates that the restore shall be done to an S3 endpoint.
+	RestoreTypeS3 = "s3"
+
+	// RestoreTypeFolder indicates that the restore shall be done to a folder,
+	// usually a RWX PVC mounted to the Pod of the restore process.
+	RestoreTypeFolder = "folder"
+)
+
+var (
+	// Config contains the values of the user-provided configuration of the operator module,
+	// combined with the default values as defined in operator.Command.
+	Config = &Configuration{}
+)
+
+// Configuration holds a strongly-typed tree of the configuration
+type Configuration struct {
+	DoCheck   bool
+	DoPrune   bool
+	DoRestore bool
+	DoArchive bool
+
+	BackupCommandAnnotation       string
+	BackupFileExtensionAnnotation string
+	BackupDir                     string
+
+	PromURL    string
+	WebhookURL string
+
+	Hostname   string
+	KubeConfig string
+
+	ResticBin        string
+	ResticRepository string
+	ResticOptions    string
+
+	RestoreDir         string
+	RestoreS3Endpoint  string
+	RestoreS3AccessKey string
+	RestoreS3SecretKey string
+	RestoreSnap        string
+	RestoreType        string
+	RestoreFilter      string
+	VerifyRestore      bool
+	RestoreTrimPath    bool
+
+	PruneKeepLast    *int
+	PruneKeepHourly  *int
+	PruneKeepDaily   *int
+	PruneKeepWeekly  *int
+	PruneKeepMonthly *int
+	PruneKeepYearly  *int
+	PruneKeepTags    bool
+
+	Tags []string
+}
+
+// Validate ensures a consistent configuration and returns an error should that not be the case
+func (c *Configuration) Validate() error {
+	if c.DoRestore {
+		c.RestoreType = strings.ToLower(c.RestoreType)
+		switch c.RestoreType {
+		case RestoreTypeS3:
+			switch {
+			case "" == c.RestoreS3Endpoint:
+				return fmt.Errorf("if the restore type is set to '%s', then the restore s3 endpoint must be defined", RestoreTypeS3)
```

Comment:
This kinda reads strange, why not
```suggestion
			case c.RestoreS3Endpoint == "":
				return fmt.Errorf("if the restore type is set to '%s', then the restore s3 endpoint must be defined", RestoreTypeS3)
```

Reply:
It comes from certain languages that require to call `equals` for String comparison, e.g. Java. There you want to do `"".equals(otherString)`. The reason is that if `otherString` is `null`, then `"".equals(otherString)` still works, whereas `otherString.equals("")` does not. This habbit sticks with me.

There are sometimes other benefits. E.g. here all the `""` align vertically, whereas they wouldn't if I'd write them the other way round. So it's easier to see that all three checks work similarly.